front/app/containers/Admin/pagesAndMenu/containers/GenericHeroBannerForm/index.tsx

```diff
@@ -45,61 +45,109 @@ type MultilocErrorType = {
   signedOutSubheaderErrors: Multiloc;
 };
 export type PreviewDevice = 'mobile' | 'tablet' | 'desktop';
+import { TBreadcrumbs } from 'components/UI/Breadcrumbs';
+
+import Outlet from 'components/Outlet';
 
 // resources
-import {
-  IHomepageSettingsAttributes,
-  updateHomepageSettings,
-} from 'services/homepageSettings';
-import useHomepageSettings from 'hooks/useHomepageSettings';
+import { IHomepageSettingsAttributes } from 'services/homepageSettings';
 
 // utils
 import { isNil, isNilOrError } from 'utils/helperUtils';
-import { isCLErrorJSON } from 'utils/errorUtils';
-import { forOwn, size, trim, debounce, isEqual } from 'lodash-es';
+import { forOwn, size, trim, debounce } from 'lodash-es';
 import { convertUrlToUploadFile } from 'utils/fileUtils';
 
 // constants
-import { pagesAndMenuBreadcrumb, homeBreadcrumb } from '../../breadcrumbs';
 import Warning from 'components/UI/Warning';
+import { ICustomPagesAttributes } from 'services/customPages';
 const TITLE_MAX_CHAR_COUNT = 45;
 const SUBTITLE_MAX_CHAR_COUNT = 90;
 
 const BgHeaderPreviewSelect = styled(Select)`
   margin-bottom: 20px;
 `;
 
-const HeroBannerForm = ({ intl: { formatMessage } }: InjectedIntlProps) => {
+// names differ slightly between HomePage and CustomPage
+type Props = {
+  type: 'homePage' | 'customPage';
+  breadcrumbs: TBreadcrumbs;
+  title?: string | JSX.Element;
+  formStatus: ISubmitState;
+  setFormStatus: (ISubmitState) => void;
+  onSave: (HeroBannerInputSettings) => void;
+  isLoading: boolean;
+  inputSettings: HeroBannerInputSettings;
+  apiErrors?: CLErrors | null;
+};
+
+export type HeroBannerInputSettings = {
```

Comment:
The types here are named after the CustomPage prop names, as those are simpler

If there's a cleaner/easier way to do this in TS please let me know